lib/base/connection.js

```diff
@@ -0,0 +1,556 @@
+  //     this._inTransaction = true
+  //     this.emit('begin')
+  //   } catch (err) {
+  //     this.emit('error', err)
+  //     throw err;
+  //   }
+  // }
+
+  // /**
+  //  * Commit a transaction.
+  //  *
+  //  * @param {basicCallback} [callback] A callback which is called after transaction has commited, or an error has occurred. If omited, method returns Promise.
+  //  * @return {Transaction|Promise}
+  //  */
+
+  // async commit() {
+  //   if (this._aborted) {
+  //     throw new TransactionError("The transaction is automatically rolled back due to a serious error, pls use `rollback()` to cancel transaction.", 'EABORT')
+  //   }
+
+  //   if (!this._inTransaction) {
+  //     throw new TransactionError('Transaction has not begun. Call begin() first.', 'ENOTBEGUN')
+  //   }
+
+  //   if (this._activeInstance) {
+  //     throw new TransactionError("Can't commit transaction. There is a request in progress.", 'EREQINPROG')
+  //   }
+
+  //   try {
+  //     debug('@Connection (%d): transaction commit', IDS.get(this))
+  //     await this._commit(this._connection)
+  //     debug('@Connection (%d): transaction commited', IDS.get(this))
+  //     this._inTransaction = false;
+  //   } catch (err) {
+  //     this.emit('commit')
+  //     throw err
+  //   }
+  // }
+
+  // /**
+  //  * Rollback a transaction.
+  //  *
+  //  * @param {basicCallback} [callback] A callback which is called after transaction has rolled back, or an error has occurred. If omited, method returns Promise.
+  //  * @return {Transaction|Promise}
+  //  */
+  // async rollback() {
+  //   if (!this._inTransaction) {
+  //     throw new TransactionError('Transaction has not begun. Call begin() first.', 'ENOTBEGUN')
+  //   }
+
+  //   if (this._activeInstance) {
+  //     throw new TransactionError("Can't rollback transaction. There is a request in progress.", 'EREQINPROG')
+  //   }
+
+  //   try {
+  //     debug('@Connection (%d): transaction rollback', IDS.get(this))
+  //     this._inTransaction = false;
+  //     if (!this._aborted) {
+  //       await this._rollback(this._connection)
+  //     }
+  //     this.emit('rollback', this._aborted)
+  //     this._aborted = undefined;
+  //     debug('@Connection (%d): transaction rolled back', IDS.get(this))
+  //   } catch (err) {
+  //     this.emit('error', err)
+  //     throw err
+  //   }
+  // }
+
+  // /**
+  //  * @private
+  //  * @param {basicCallback} [callback]
+  //  * @return {Transaction}
+  //  */
+
+  // // eslint-disable-next-line no-unused-vars
+  // _beginTrans(tedious, isolationLevel) {
+  //   throw new Error('Not implementation.')
+  // }
+
+  // // eslint-disable-next-line no-unused-vars
+  // _commit(tedious) {
+  //   throw new Error('Not implementation.')
+  // }
+
+  // // eslint-disable-next-line no-unused-vars
+  // _rollback(tedious) {
+  //   throw new Error('Not implementation.')
+  // }
+
+}
+
+Connection.defaultIsolationLevel = ISOLATION_LEVEL.READ_COMMITTED
```

Comment:
This prop doesn't appear to be used anywhere - should it not be set on a per instance basis anyway?

Reply:
Sorry, I have bad habits. I'll correct them.